tickets.js

```diff
@@ -0,0 +1,155 @@
+var Users = [];
+var tickets = [];
+const TicketCategories = ["Bug", "Feature", "Support"];
+const unresolvedTickets = [];
+
+function CreateUser(userName, Email) {
+  const user = { id: Users.length + 1, name: userName, email: Email };
+  Users.push(user);
+  console.log("User created:", userName);
+  return user;
+}
+
+function createTicket(title, description, category, priority, userId) {
+  const ticket = {
+    id: tickets.length + 1,
+    title: title,
+    description: description,
+    category: category,
+    priority: priority,
+    status: "Open",
+    assignedTo: userId
+  };
+  tickets.push(ticket);
+  console.log(`Ticket created with ID: ${ticket.id}`);
+  return ticket;
+}
+
+function displayTickets() {
+  console.log("Displaying all tickets...");
+  tickets.forEach(ticket => {
+    console.log("Ticket ID:", ticket.id, "Title:", ticket.title, "Status:", ticket.status);
+  });
+}
+
+function findTicketsByUser(userId) {
+  const userTickets = tickets.filter(ticket => ticket.assignedTo === userId);
+  if (userTickets.length > 0) {
+    console.log(`Tickets for User ID ${userId}:`);
+    userTickets.forEach(ticket => {
+      console.log("Ticket ID:", ticket.id, "Title:", ticket.title, "Status:", ticket.status);
+    });
+  } else {
+    console.log("No tickets found for this user.");
+  }
+}
+
+function updateTicketStatus(ticketId, newStatus) {
+  const ticket = tickets.find(t => t.id === ticketId);
+  if (ticket) {
+    ticket.status = newStatus;
+    console.log(`Ticket ID ${ticketId} updated to ${newStatus}`);
+  } else {
+    console.log("Ticket not found.");
+  }
+}
+
+function AssignTicket(ticketId, userId) {
+  const ticket = tickets.find(t => t.id === ticketId);
+  if (ticket) {
+    ticket.assignedTo = userId;
+    console.log(`Ticket ID ${ticketId} assigned to User ID ${userId}`);
+  } else {
+    console.log("Ticket not found.");
+  }
+}
+
+function DisplayOpenTickets() {
+  const openTickets = tickets.filter(t => t.status === "Open");
+  console.log("Open Tickets:");
+  openTickets.forEach(ticket => {
+    console.log("Ticket ID:", ticket.id, "Title:", ticket.title, "Priority:", ticket.priority);
+  });
+}
+
+function CloseTicket(ticketId) {
+  const ticket = tickets.find(t => t.id === ticketId);
+  if (ticket) {
+    ticket.status = "Closed";
+    console.log(`Ticket ID ${ticketId} is now closed`);
+  } else {
+    console.log("Ticket not found.");
+  }
+}
+
+function ResolveTicket(ticketId) {
+  const ticket = tickets.find(t => t.id === ticketId);
+  if (ticket) {
+    ticket.status = "Resolved";
+    console.log(`Ticket ID ${ticketId} marked as resolved`);
+  } else {
+    console.log("Ticket not found.");
+  }
+}
+
+function viewUnresolvedTickets() {
+  const unresolvedTickets = tickets.filter(t => t.status !== "Resolved" && t.status !== "Closed");
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Variable shadowing in viewUnresolvedTickets function</b></div>
<div id="fix">

The `unresolvedTickets` constant shadows the global array of the same name. Consider renaming the local variable to avoid confusion.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -96 +96 @@
 -  const unresolvedTickets = tickets.filter(t => t.status !== "Resolved" && t.status !== "Closed");
 +  const pendingTickets = tickets.filter(t => t.status !== "Resolved" && t.status !== "Closed");
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/BitoPythonTest/pull/21#issuecomment-2456310783>#f099f2</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged